rpc_differ/rpc_differ.py

```diff
@@ -111,19 +112,25 @@ def create_parser():
     return parser
 
 
-def get_osa_commits(repo_dir, old_commit, new_commit):
-    """Get OSA commits from the RPC repository."""
-    repo = Repo(repo_dir)
-
-    repo.head.reference = repo.commit(old_commit)
-    repo.head.reset(index=True, working_tree=True)
-    old_osa_commit = repo.submodules['openstack-ansible'].hexsha
-
-    repo.head.reference = repo.commit(new_commit)
+def get_osa_commit(repo, ref):
+    """Get the OSA sha referenced by an RPCO Repo."""
+    repo.head.reference = repo.commit(ref)
     repo.head.reset(index=True, working_tree=True)
-    new_osa_commit = repo.submodules['openstack-ansible'].hexsha
-
-    return (old_osa_commit, new_osa_commit)
+    try:
+        osa_commit = repo.submodules['openstack-ansible'].hexsha
+    except IndexError:
+        # This branch doesn't use a submodule for OSA
+        # Pull the SHA out of functions.sh
+        quoted_re = re.compile('OSA_RELEASE:-"([^"]+)"')
+        functions_path = \
+            "{}/scripts/functions.sh".format(repo.working_tree_dir)
+        with open(functions_path, "r") as funcs:
+            for line in funcs.readlines():
+                match = quoted_re.search(line)
+                if match:
+                    osa_commit = match.groups()[0]
+                    break
```

Comment:
I'd suggest tacking an `else` block on the end of the `for` so the function can fail with a useful message if this pattern is never matched. A `NameError` is not that useful.

Reply:
Thanks for reviewing, addressed in 4d36dd1